### PR TITLE
Allow exporting seed in taker UI

### LINKED ISSUE #1065: Make it easier to back up the seed-file
After our discussion in https://github.com/itchysats/itchysats/issues/865 and #1016 we dropped the feature to backup and re-initialize the ItchySats (wallet) seed.

Allow users of the binary to backup their seed in textual form (so they don't have to backup a complete file, the generated seed-file is still binary), so they can save it in e.g. a password safe.

### PR DESCRIPTION
**Pull request by klochowicz**
TODO: 
- [x] Restoring from the seed (will be covered by #2989 )
- [x]  Encrypting the saved seed (or at least being clear that it needs to be handled safely) (will be covered by #897 )
- [x] A toast message about the outcome of the request (success or fail)
- ~~Add tests~~

### CODE REVIEW

File: taker/src/routes.rs [hunk 1]
```diff
@@ -336,6 +340,15 @@ pub async fn get_version() -> Json<HealthCheck> {
     })
 }
 
+#[rocket::get("/seed_backup")]
+#[instrument(name = "GET /seed_backup", skip_all)]
+pub async fn get_seed_backup(
+    seed: &State<Arc<ThreadSafeSeed>>,
```
Comment (bonomat) [inline taker/src/routes.rs:349]: Don't forget to add the `user` guard, otherwise this API does not need authentication.

**Comment by holzeis** [comment]
Added a toast about the successful outcome. file-saver will not produce any errors as it is based on the native html 5 download href, hence no error message in case of no success.

**Comment by holzeis** [comment]
Opted against adding regression tests, as there is not much logic to test anyway.

**Review by bonomat — APPROVED** [review]
Haven't tested it but looks good to me. Can you squash the commits please?

**Comment by klochowicz** [comment]
I'm not sure whether this should go in as-is (hence the draft status), especially without prior to testing this. I have kept it as draft as I noticed on Friday that the saved file is not the same binary size as the original one; and worked on a test that would ensure we backup the right thing.

 I suspect RocketResponderPro is adding headers / screwing up encoding - might be easier to go with serde and then deserialise into a blob to ensure the file is as it's supposed to be. 

The other way (perhaps simpler altogether) that I came up with during the weekend (but haven't tried) would be just serving `taker_seed` as a `NamedFile` in Rocket.  Of course then we can't backup from `--app-seed`, but that's not a requirement.

**Comment by klochowicz** [comment]
> Opted against adding regression tests, as there is not much logic to test anyway.

As this is one of the paths where the user might lose money if it doesn't work properly, I would certainly test this to prove that it works.
However, manual testing should be enough for this PR, don't worry about it here - I already started working on the test, so I will submit it separately for review 😊

**Comment by holzeis** [comment]
> I'm not sure whether this should go in as-is (hence the draft status), especially without prior to testing this. I have kept it as draft as I noticed on Friday that the saved file is not the same binary size as the original one; and worked on a test that would ensure we backup the right thing.

The downloaded seed file works properly.. I've tested it manually.

1. startup itchysats
2. backup seed
3. stop itchysats
4. replace seed file with downloaded one
5. startup itchysats

Everything is working ;)

>> Opted against adding regression tests, as there is not much logic to test anyway.

>As this is one of the paths where the user might lose money if it doesn't work properly, I would certainly test this to prove that it works.

I think the regressions tests are better covered including the restore feature.

> The other way (perhaps simpler altogether) that I came up with during the weekend (but haven't tried) would be just serving taker_seed as a NamedFile in Rocket. Of course then we can't backup from --app-seed, but that's not a requirement.

I guess that would also work, for the app-seed argument I was thinking if we could store that also to a taker_seed file if we get it, then we have the same behavior. wdyt?

**Comment by klochowicz** [comment]
> The downloaded seed file works properly.. I've tested it manually.
> 
> 1. startup itchysats
> 2. backup seed
> 3. stop itchysats
> 4. replace seed file with downloaded one
> 5. startup itchysats
> 
> Everything is working ;)

awesome! 🤘

**Comment by klochowicz** [comment]
> I guess that would also work, for the app-seed argument I was thinking if we could store that also to a taker_seed file if we get it, then we have the same behavior. wdyt?

that would also work, yes! however it might be a too niche feature to even bother. :-)

**Pull request merged**